client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue

```diff
@@ -325,6 +366,61 @@ export default {
       return null
     },
 
+    statementSubmitterField () {
+      const attr = this.localStatement.attributes
+      let submitterField = 'authorName'
+      // If submitter is an orga and name has a value
+      if (attr.submitName && !attr.isSubmittedByCitizen) {
+        submitterField = 'submitName'
+      }
+
+      return submitterField
+    },
+
+    statementSubmitterValue: {
+      get () {
+        const attr = this.localStatement.attributes
+        let submitterValue = attr[this.statementSubmitterField]
+        // If submitter has revoked the gdpr consent for this statement or statement has been anonymized
+        if (this.isSubmitterAnonymous()) {
+          submitterValue = Translator.trans('anonymized')
+        }
+
+        return submitterValue
+      },
+      set (value) {
+        this.localStatement.attributes[this.statementSubmitterField] = value
+      }
+    },
+
+    submitterHelpText () {
+      const attr = this.localStatement.attributes
+      let helpText = ''
+
+      if (attr.gdprConsent && attr.gdprConsent.consentRevoked) {
+        helpText = Translator.trans('personal.data.usage.revoked')
+        if (hasPermission('area_statement_anonymize') &&
+          attr.original.submitterAndAuthorMetaDataAnonymized) {
+          helpText = helpText + `<br><br>${Translator.trans('statement.anonymized.submitter.data')}`
+        }
+      } else if (hasPermission('area_statement_anonymize') &&
+        attr.original.submitterAndAuthorMetaDataAnonymized) {
+        helpText = Translator.trans('statement.anonymized.submitter.data')
+      }
+
+      return helpText
+    },
+
```

Comment:
nitpick: how about doing it like this? most importantly, I'd use destructuring for the attributes and store the conditions in variables so it is easy to understand what the conditions are. Also, I like to make conditions explicit (instead of using `else`), at least if the conditions are a little more complex.
```suggestion
    submitterHelpText () {
            const { gdprConsent, original } = this.localStatement.attributes
      let helpText = ''

      const isConsentRevoked = gdprConsent?.consentRevoked
      const isAnonymized = hasPermission('area_statement_anonymize') && original.submitterAndAuthorMetaDataAnonymized

      if (isConsentRevoked) {
        helpText = Translator.trans('personal.data.usage.revoked')

        if (isAnonymized) {
          helpText = `${helpText} <br><br>${Translator.trans('statement.anonymized.submitter.data')}`
        }
      }

      if (!isConsentRevoked && isAnonymized) {
        helpText = Translator.trans('statement.anonymized.submitter.data')
      }
      
      return helpText
    },

```

Reply:
I like this approach, thanks. Done with [303a75a](https://github.com/demos-europe/demosplan-core/pull/2899/commits/303a75ac1c75fdc28c19a6754edb92b4aa032693)